deployment_scripts/deploy_linux.sh

```diff
@@ -101,25 +101,25 @@ log_message "Creating island dirs under $ISLAND_PATH"
 mkdir -p "${MONGO_PATH}" || handle_error
 mkdir -p "${ISLAND_BINARIES_PATH}" || handle_error
 
-# Detecting command that calls python 3.7
+# Detecting command that calls python 3.11
 python_cmd=""
-if [[ $(python --version 2>&1) == *"Python 3.7"* ]]; then
+if [[ $(python --version 2>&1) == *"Python 3.11."* ]]; then
   python_cmd="python"
 fi
-if [[ $(python37 --version 2>&1) == *"Python 3.7"* ]]; then
-  python_cmd="python37"
+if [[ $(python311 --version 2>&1) == *"Python 3.11."* ]]; then
+  python_cmd="python311"
 fi
-if [[ $(python3.7 --version 2>&1) == *"Python 3.7"* ]]; then
-  python_cmd="python3.7"
+if [[ $(python3.11 --version 2>&1) == *"Python 3.11."* ]]; then
+  python_cmd="python3.11"
 fi
 
 if [[ ${python_cmd} == "" ]]; then
-  log_message "Python 3.7 command not found. Installing python 3.7."
+  log_message "Python 3.11 command not found. Installing python 3.11."
   sudo add-apt-repository ppa:deadsnakes/ppa
   sudo apt-get update
-  sudo apt-get install -y python3.7 python3.7-dev python3.7-venv
-  log_message "Python 3.7 is now available with command 'python3.7'."
-  python_cmd="python3.7"
+  sudo apt-get install -y python3.11 python3.11-dev python3.11-venv
+  log_message "Python 3.11 is now available with command 'python3.11'."
+  python_cmd="python3.11"
```

Comment:
If we create a variable to contain "3.11", then upgrading this in the future would be a single-line change.